bin/install.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+# This script downloads the latest release version of EAS CLI from GitHub and
+# installs it in /usr/local/bin
+
+{
+  # All code is inside a block to ensure the script executes only when
+  # downloaded completely.
+
+  set -e
+
+  prefix="${EAS_PREFIX:-/usr/local}"
```

Comment:
I'm not sure about macOS, but on Linux scripts that are installed this way always default to somewhere in home dir. It looks very suspicious when a script like that requires `sudo`.

I'm not sure If we want to keep the same behavior between platform, but I think that for Linux it would be better to 
- copy all code to `~/.eas` 
- make sure that line `export PATH=$HOME/.eas/bin:$PATH` exists in `.bashrc` and `~/.zshrc`

or instead of adding it PATH we could run some other script (e.g. the same as nvm)
```
export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"
```

Reply:
We wanted to avoid automatically modifying .bashrc/.zshrc because it's fragile to update and people may use multiple shells. On macOS, `/usr/local/{lib,bin}` are set up by Homebrew with the primary user's own permissions, so they should be writeable most of the time.

Reply:
> fragile to update 

We need only to make sure that one specific line is present and if not add that to the end of the file

>  people may use multiple shells

If `rc` file exists we could add there automatically(.barc and .zshrc is probably enough) and for other shells, we could print the message "Make sure to add this line to you rc file.."

I think that requiring `sudo` is acceptable only with a package supported by distro(`deb`, `rpm`, ...) where it's clear how it works and how to uninstall it or update. If I saw install script doing sth like this I wouldn't use it if had a choice, I don't think that this will be a deal-breaker for most of the users, but I think it's a pretty significant red flag.

Reply:
I've had problems with tools that use this profile modification approach in the past. E.g. we only add it to `.zshrc` or `.zprofile` because the user is using Zsh and/or because those files exist. Now if the use (even temporarily) switches to Bash (sometimes it's a script written by someone else, possibly running inside a build phase of Xcode, in a VS Code plugin, possibilities are endless..), the user will just get an error like `bash: eas: command not found` and it's not obvious at all why this is happening or what they should do to fix it.

There are many files too: `.profile`, `.bashrc`, `.bash_profile`, `.zshrc` and `.zprofile`... just for Bash and Zsh. Which one would we pick? Search in all of them in case the user previously added the snippet manually?

Additionally, I wouldn't copy nvm too closely: not sure how it does it but its startup script adds a visible delay to opening terminal Windows which is why I switched away from nvm a long time ago.

These are tradeoffs, but since I was aiming to create a single-command, reliable way to install the CLI, having to sometimes type your password seems less bad than having the script randomly fail or require manual steps. The current approach also works well in CI systems where the code is already running as root.

Reply:
I agree that interacting with `rc`/`profile` file is not ideal, but (in case of linux) this is a best way if we want to have both simple install script and seamless autoupdate.

- proper solution would be to have distro specific packages, but it's probably not something we have time right now.
- current solution meets both conditions, but you are assigning ownership to non root user for files in `/usr/local` which looks very sketchy.
- If we would go with current solution, but also `chown` files as root, it would be a lot better, it's still no ideal and we lose easy auto update, because password would be required every time, but it's probably good enough. If you feel strongly against bashrc/zshrc approach I would suggest to go this way.


For all non linux cases I agree that this approach is perfectly fine, with one issue I mentioned in other comment. You are assuming that user will always have uid=1000(that is how `tar` stores ownership info), which may not be always true.

Reply:
I think we will be able to create at least Ubuntu/Debian packages (.deb) with relative ease a bit later, if users ask for it, because oclif dev cli has support for this out of the box (https://oclif.io/docs/releasing#ubuntudebian-packages). These won't have our automatic updates, but I think it's fair to assume that a user installing using that method wants to use the package manager of the distro to also handle updates, so that's fine.

Owning the files as root doesn't break automatic updates because oclif has built in support for them in the executable that looks for the updates in the home dir. So we could `chown` the files as root, perhaps we should do that at least on Linux. I'm still thinking whether we should do that on macOS too - always running the script with sudo would make things simpler in many ways. (If the user has permissions to `/usr/local` that adds a bit of unnecessary friction, but we can't rely on having these permissions anyway.

Reply:
>  it's fair to assume that a user installing using that method wants to use the package manager of the distro to also handle updates

Definitely -- it would be confusing if apt/dnf thinks the package is v1.0.0 but the automatic updater actually upgraded the package to v1.0.1.

> So we could chown the files as root, perhaps we should do that at least on Linux.

Perhaps the cleanest heuristic to is to `chown` eas-cli's files to match the user + group of the parent directory (`/usr/local/bin` and `/usr/local/lib`). On macOS with a normal Homebrew installation (I set up a new computer a few months ago), this looks like `$me:admin` for `bin` and `$me:staff` for `lib`.